Intelligentgraph

peterjohnlawrence opened the pull request:
Fixes #111

aeberhart reviewed (changes requested):
Thanks for the PR. Almost there, we just need to move the path test out of the regular proj / emp tests

peterjohnlawrence commented:
- Restored DBTest, 
- Removed testPath from RDF4JTest, 
- Created RDF4JPathQLTest, 
- Added more test data in junit.additions.nt, 
- Created a new service in dj/pathql.json

In RDF4JPathQLTest,  I overrode some tests from DBTest, but these will refer to the 'incorrect' db, rather than the new service database (dj/pathql)

aeberhart commented on dashjoin-rdf4j/src/test/java/org/dashjoin/service/RDF4JPathQLTest.java:
```diff
@@ -0,0 +1,126 @@
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class RDF4JPathQLTest extends DBTest {
```
You don't need to extend DBTest. You can simply annotate the class with QuarkusTest and simply add

  @Inject
  Services services;

  @Inject
  Data db;

peterjohnlawrence replied:
... but then missing a lot of supporting methods in DBTest such as toID() etc.

Do I need to add additional tests other than testPath()? I only added them because they would fail in DBTest when dj/junit db was used with both junit.nt and junit.additions.nt

aeberhart commented on dashjoin-rdf4j/src/main/java/org/dashjoin/service/RDF4J.java:
```diff
@@ -499,27 +512,74 @@ void add(Table table, Map<String, Object> res, IRI predicate, Value value, boole
-      _cp = new SPARQLRepository(endpoint);
-      ((SPARQLRepository) _cp).setUsernameAndPassword(username, password);
-    }
+	  IntelligentGraphConfig intelligentGraphConfig = null;
```
we use the default eclipse java formatter. Would be good to keep the style consistent

peterjohnlawrence replied:
Not sure I understand, I *think* I use the default formatter as well.

aeberhart commented on dashjoin-rdf4j/src/test/java/org/dashjoin/service/RDF4JTest.java:
```diff
@@ -54,9 +56,41 @@ public void testGetTables() throws Exception {
-  public void testPath() throws Exception {
-    // SPARQL does not support tracking property paths
-  }
+//  @Override
```
I think we need the empty override. Otherwise, I get a SPARQL parse error since OpenCypher query from DBTest is fed. Note that the RDF4JTest is not part of the CICD test on GitHub yet
